telescope/src/main/java/com/mattprecious/telescope/TelescopeProjectionService.java

```diff
@@ -0,0 +1,45 @@
+package com.mattprecious.telescope;
+
+import android.annotation.TargetApi;
+import android.app.Notification;
+import android.app.NotificationChannel;
+import android.app.NotificationManager;
+import android.app.Service;
+import android.content.Intent;
+import android.os.IBinder;
+import android.support.annotation.Nullable;
+
+import static android.os.Build.VERSION_CODES.Q;
+
+@TargetApi(Q)
+public class TelescopeProjectionService extends Service {
+    public static final String NOTIFICATION_CHANNEL_ID = "Telescope Notifications";
+    public static final int SERVICE_ID = NOTIFICATION_CHANNEL_ID.hashCode();
+
+    @Override public void onCreate() {
+        super.onCreate();
+
+        createNotificationChannel();
+        startForeground(
+                SERVICE_ID,
+                new Notification.Builder(this, NOTIFICATION_CHANNEL_ID).build()
+        );
+    }
+
+    private void createNotificationChannel() {
+        NotificationChannel serviceChannel = new NotificationChannel(
+                NOTIFICATION_CHANNEL_ID,
+                "Telescope Service Channel",
+                NotificationManager.IMPORTANCE_DEFAULT
+        );
+
+        NotificationManager notificationManager = getSystemService(NotificationManager.class);
+        notificationManager.createNotificationChannel(serviceChannel);
```

Comment:
Is this safe to do repeatedly? Or do we need to check and no-op if it's already present?

Reply:
I believe it's safe and not worth the extra-check. We do create notification channels on our application startup and haven't seen issues so far.